### DISCUSSION TITLE
No auto scrolling while streaming

### DISCUSSION THREAD

**Discussion by isentropic** [olimorris/codecompanion.nvim#302] [Ideas]
Not sure if its even possible in nvim, but is it possible to disable the scrolling automatically? I can't keep up with the token generation but I also can't freeze the cursor position. Meaning that with long answers I need to wait for the end of the response.
Not sure if its nvim limitation or something. Great plugin btw, best in the market for nvim

**Comment by olimorris** [comment]
If you move your cursor when it streams it will disable auto scrolling.

I'll be happy to add this in as a config option. I too struggle to keep up with some LLMs

**Comment by znxuz** [comment]
maybe you can also use hooks like these for pushing the cursor line to the top and trigger cursor change so that it doesn't follow the stream:

```lua
local autocmd_group = vim.api.nvim_create_augroup("CodeCompanionHooks", {})

vim.api.nvim_create_autocmd({ "User" }, {
	pattern = "CodeCompanionRequestStarted",
	group = autocmd_group,
	callback = function(request)
		if request.data.strategy == "chat" then
			vim.api.nvim_feedkeys('zt', 'n', false)
		end
	end,
})

vim.api.nvim_create_autocmd({ "User" }, {
	pattern = "CodeCompanionRequestStreaming",
	group = autocmd_group,
	callback = function(request)
		if request.data.strategy == "chat" then
			vim.api.nvim_feedkeys('H', 'n', false)
		end
	end,
})
```

somehow the second hook doesn't fire or maybe fires a bit too early when you send a request inline and it turns into a question copied into a chat buffer, but for chat buffer it works fine for me at least.
